packages/shared/src/components/MainLayout.tsx

```diff
@@ -176,8 +176,7 @@ function MainLayoutComponent({
 
   if (
     (!isPageReady && isPageApplicableForOnboarding) ||
-    shouldRedirectOnboarding ||
-    !isAuthReady
```

Comment:
I am not 100% sure why `!isAuthReady` was added in https://github.com/dailydotdev/apps/pull/3675, but don't see it breaks anything currently.

Reply:
Not sure but this now breaks:
![Screenshot 2024-10-22 at 12 20 05](https://github.com/user-attachments/assets/906637ab-cb18-4617-b3ef-a78620371266)

Also when testing on metatags.io still not seeing anything for posts and posts/[id]

Reply:
@rebelchris for metatags.io make sure to visit the page once so it pre-renders on demand (because we use `blocking: false` and then check which will show metadata. Next.js probably does not see metatags.io user agent as valid SEO bot so it returns non blocking (empty page) for them. 

As to sidebar issue, yeah it was probably something with `isAuthReady`, I asked on the PR why it was added, worst case we can revert whole boot optimization PR to see if it fixes sidebar.

Reply:
It all rooted to the hydration issue. Now that we have less re-renders(which acts as the hydrated use state), we get hydration issues.

Reply:
Yeah thanks, managed to figure that out and fixed them. Was just worried it was something else.

But yeah, in general, while its easy to fix hydration issues by just not rendering in SSR we have to be careful since it can affect SEO like yesterday. So its always good to test those cases as well.